src/main/java/com/redislabs/redisai/Tensor.java

```diff
@@ -30,4 +90,22 @@ public DataType getDataType() {
   public void setDataType(DataType dataType) {
     this.dataType = dataType;
   }
+
+  /**
+   * Encodes the current model properties into an AI.MODELSET command to be store in RedisAI Server
+   *
+   * @param key name of key to store the Model
+   * @return
+   */
+  protected List<byte[]> getTensorSetCommandBytes(String key) {
+    List<byte[]> args = new ArrayList<>();
```

Comment:
@filipecosta90 I would have try to optimize this method.
1. Do you know the args length in advance?
2. Can you replace Arraylist<byte[]> with byte[][]?

Reply:
for:
1. we can calculate it but makes the code more obscure IMHO.
2. we can make it, but i really believe that it makes the code less clear:
```
   int arraylen = 1;
    int arraypos = 0;
    for (long shapeDimension : shape) {
      arraylen*=shapeDimension;
    }
    arraylen+=3+shape.length;

    byte[][] args = new byte[arraylen][];
    args[0]=SafeEncoder.encode(key);
    args[1]=dataType.getRaw();
    arraypos=2;
    for (long shapeDimension : shape) {
      args[arraypos]=Protocol.toByteArray(shapeDimension);
      arraypos++;
    }
    args[arraypos]=Keyword.VALUES.getRaw();
    for (byte[] encodedElem : dataType.toByteArray(values, shape)) {
      args[arraypos]=encodedElem;
      arraypos++;
    }
```
I would recommend us to keep the List of bytes guy. wdyt?